harmony_service/adapter.py

```diff
@@ -0,0 +1,78 @@
+"""`HarmonyAdapter` for SMAP-L2-Gridding-Service.
+
+The class in this file is the top level of abstraction for a service that will
+accept an input L2G input granule and transform it into a fully gridded
+NetCDF-CF output file
+
+Currently, the service works with SPL2SMP_E granules.
+
+"""
+
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+from harmony_service_lib import BaseHarmonyAdapter
+from harmony_service_lib.message import Source as HarmonySource
+from harmony_service_lib.util import download, generate_output_filename, stage
+from pystac import Asset, Item
+
+from smap_l2_gridder.grid import transform_l2g_input
+
+
+class SMAPL2GridderAdapter(BaseHarmonyAdapter):
+    """Custom adapter for SMAP-L2-Gridder Service."""
+
+    def process_item(self, item: Item, source: HarmonySource) -> Item:
+        """Process single input STAC item."""
+        with TemporaryDirectory() as working_directory:
+            try:
+                results = item.clone()
+                results.assets = {}
+
+                asset = next(
+                    item_asset
+                    for item_asset in item.assets.values()
+                    if 'data' in (item_asset.roles or [])
+                )
+
+                # Download the input:
+                input_filepath = download(
+                    asset.href,
+                    working_directory,
+                    logger=self.logger,
+                    cfg=self.config,
+                    access_token=self.message.accessToken,
+                )
+
+                working_filename = Path(input_filepath).parent / 'working_gridded.nc'
+
+                final_target_filename = generate_output_filename(
+                    asset.href, is_regridded=True, ext='.nc'
```

Comment:
Is that the standard extension for netcdf4 files?  I'm fine changing it if so.

turns out it is not. 

[The recommended file extension for netCDF-4, and the default supplied by the Unidata software library is .nc, the same as for netCDF-3](https://www.loc.gov/preservation/digital/formats/fdd/fdd000332.shtml#:~:text=The%20recommended%20file%20extension%20for,be%20used%20in%20some%20contexts.). However, since the files are valid HDF5 files, the .h5 extension may be used in some contexts. [See Why aren't different extensions used for the different [netCDF] formats, for example, .nc3 and .nc4?](https://docs.unidata.ucar.edu/netcdf-c/current/faq.html#fv17)

Reply:
Okay, I must admit that the:

> the same as for netCDF-3

is exactly what I wanted to avoid, but this is _the_ reference to follow, so I'm fine leaving it as-is.